diff -up firefox-128.0/browser/app/profile/firefox.js.addons firefox-128.0/browser/app/profile/firefox.js
--- firefox-128.0/browser/app/profile/firefox.js.addons	2024-06-13 11:47:04.255428350 +0200
+++ firefox-128.0/browser/app/profile/firefox.js	2024-06-13 11:53:00.442837371 +0200
@@ -56,7 +56,8 @@ pref("extensions.systemAddon.update.enab
 
 // Disable add-ons that are not installed by the user in all scopes by default.
 // See the SCOPE constants in AddonManager.sys.mjs for values to use here.
-pref("extensions.autoDisableScopes", 15);
+pref("extensions.autoDisableScopes", 0);
+pref("extensions.showMismatchUI", false);
 // Scopes to scan for changes at startup.
 pref("extensions.startupScanScopes", 0);
 
